### PR TITLE
feature: publish ProjectsPanel

### PR DESCRIPTION
**Pull request by gcor**
Our ui components library. With storybook and npm package

storybook — https://ekb-ui.vercel.app/
npm package —  https://www.npmjs.com/package/ekb package

How to use
```ts
npm i ekb

// in the app
import { ProjectsPanel } from 'ekb';

// ProjectsPanel props
// full types https://github.com/ekaterinburgdev/ekaterinburg-ui/blob/3d3efa0dc1bf2f16a1156d56b8749a20695dd302/src/lib/components/ProjectsPanel/types.ts
 {
  activeProjectId?: Project["id"];
  projects?: Project[];
  theme?: Theme;
  isDarkIcons?: boolean;
  style?: Styles;
  toggleStyle?: ToggleStyles;
}
```

PR in Transport: https://github.com/ekaterinburgdev/transport/pull/18/files
PR in Map: https://github.com/ekaterinburgdev/map/pull/12/files
PR in Guides: https://github.com/ekaterinburgdev/guides/pull/54/files

For @a-chabin 
Please add to your vercel our storybook instead of my https://ekb-ui.vercel.app/
Build rule: `pnpm build-storybook`

### CODE REVIEW

**Comment by a-chabin** [comment]
Let's add Iset Sans?
You can even temporarily @font-face with ekaterinburg.design

Direct links:
- https://ekaterinburg.design/fonts/IsetSans-Light.woff2
- https://ekaterinburg.design/fonts/IsetSans-Medium.woff2
- https://ekaterinburg.design/fonts/IsetSans-SemiBold.woff2
- https://ekaterinburg.design/fonts/IsetSans-Regular.woff2

image

P.S. I hope there won't be any problems with the HTTP-headers!

**Comment by a-chabin** [comment]
It seems that there will be a lot of responsive rules for this component.

How about replacing paddings and margins in `px` with em?

File: src/lib/components/ProjectsPanel/components/ProjectsPanel.module.css [hunk 1]
```diff
@@ -0,0 +1,100 @@
+.toggle {
+  display: inline-flex;
+  padding: 8px;
+  transition: 0.2s all ease;
```
Comment (a-chabin) [inline src/lib/components/ProjectsPanel/components/ProjectsPanel.module.css:4]: It seems that we can try 150ms, especially on smaller screens. However, this is a taste :)

File: src/lib/components/ProjectsPanel/components/Project.module.css [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  border-radius: 32px;
+  transition: 0.2s all ease;
+  outline: none;
+}
+
+.project:hover {
+  background-color: rgba(155, 170, 190, 0.2);
+}
+
+.project:focus,
```
Comment (a-chabin) [inline src/lib/components/ProjectsPanel/components/Project.module.css:17]: Let's add focus/focus-visible for items as logo or project 

image

image

File: src/lib/components/ProjectsPanel/components/ProjectsPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+const defaultStyle = { bottom: "16px", left: "16px", zIndex: 1000 };
+
+export interface Props {
+  activeProjectId?: Project["id"];
+  projects?: Project[];
+  theme?: Theme;
+  isDarkIcons?: boolean;
+  style?: Styles;
+  toggleStyle?: ToggleStyles;
```
Comment (a-chabin) [inline src/lib/components/ProjectsPanel/components/ProjectsPanel.tsx:21]: Maybe combine these items?

```
theme: {
  isDarkIcons?: boolean;
  style?: Styles;
  toggleStyle?: ToggleStyles;
}
```

File: src/lib/components/ProjectsPanel/components/ProjectsPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  color: "white",
+};
+
+const defaultStyle = { bottom: "16px", left: "16px", zIndex: 1000 };
+
+export interface Props {
+  activeProjectId?: Project["id"];
+  projects?: Project[];
+  theme?: Theme;
+  isDarkIcons?: boolean;
```
Comment (a-chabin) [inline src/lib/components/ProjectsPanel/components/ProjectsPanel.tsx:19]: Do we really need the word `is` in React Props?)

File: src/lib/components/ProjectsPanel/components/Modal.module.css [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  margin-left: 16px;
+  margin-right: 16px;
+  position: relative;
+  top: 8px;
+}
+
+.modal__subcontent {
+  position: absolute;
+  right: 16px;
+  bottom: -2px;
```
Comment (a-chabin) [inline src/lib/components/ProjectsPanel/components/Modal.module.css:56]: image

It seems we need to align this vertically centered

File: src/lib/components/ProjectsPanel/components/Modal.module.css [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+  line-height: var(--footer-height);
+}
+
+.modal__subcontent a {
+  margin: 0 16px;
+  font-size: 32px;
```
Comment (a-chabin) [inline src/lib/components/ProjectsPanel/components/Modal.module.css:62]: Let's try to make these `font-sizes` equivalent

- Common CSS variable with font value
- Make common parent with visible and hidden areas

image

File: src/lib/components/ProjectsPanel/components/ProjectsPanel.module.css [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+.activeproject {
+  display: flex;
+  align-items: center;
+  transition: 0.2s all;
+  border-radius: 100px;
+  padding-right: 24px;
+}
+.activeproject:not(:hover) {
+  /* use theme inside the ProjectsPanel.tsx */
+  background-color: transparent !important;
```
Comment (AtkishkinVlad) [inline src/lib/components/ProjectsPanel/components/ProjectsPanel.module.css:83]: **question** Is it possible remove `!important` in this place ?

File: src/lib/components/ProjectsPanel/components/Modal.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+import FocusLock from "react-focus-lock";
+import styles from "./Modal.module.css";
+import { Styles, Theme } from "../types";
+
+export function useModal() {
+  const [isOpen, setisOpen] = useState(false);
+  const close = useCallback(() => setisOpen(false), []);
+  const toggle = useCallback(() => setisOpen(!isOpen), [setisOpen, isOpen]);
+
+  useEffect(() => {
```
Comment (AtkishkinVlad) [inline src/lib/components/ProjectsPanel/components/Modal.tsx:12]: **suggestion** Maybe separate in `useEsc()` hook ? (maybe necessary in other components)

**Comment by AtkishkinVlad** [comment]
**suggestion** Maybe we'll try to redesign scroll ? (windows 10 example)
image

File: src/lib/components/ProjectsPanel/components/ProjectsPanel.module.css (hunk 1 above)
Comment (gcor) [reply src/lib/components/ProjectsPanel/components/ProjectsPanel.module.css]: 50ms is too small difference)

File: src/lib/components/ProjectsPanel/components/Project.module.css (hunk 2 above)
Comment (gcor) [reply src/lib/components/ProjectsPanel/components/Project.module.css:18]: added

File: src/lib/components/ProjectsPanel/components/ProjectsPanel.tsx (hunk 3 above)
Comment (gcor) [reply src/lib/components/ProjectsPanel/components/ProjectsPanel.tsx:32]: * isDarkIcons — set of icons that designers will update soon
* style — position of toggle\button and modal
* toggleStyle — only toggle style (like width for guides)

File: src/lib/components/ProjectsPanel/components/ProjectsPanel.tsx (hunk 4 above)
Comment (gcor) [reply src/lib/components/ProjectsPanel/components/ProjectsPanel.tsx:30]: I mark with `is` boolean props)

File: src/lib/components/ProjectsPanel/components/Modal.module.css (hunk 6 above)
Comment (gcor) [reply src/lib/components/ProjectsPanel/components/Modal.module.css:66]: will redesign with new icons

File: src/lib/components/ProjectsPanel/components/Modal.module.css (hunk 5 above)
Comment (gcor) [reply src/lib/components/ProjectsPanel/components/Modal.module.css]: removed)

File: src/lib/components/ProjectsPanel/components/ProjectsPanel.module.css (hunk 7 above)
Comment (gcor) [reply src/lib/components/ProjectsPanel/components/ProjectsPanel.module.css]: yep, removed

File: src/lib/components/ProjectsPanel/components/Modal.tsx (hunk 8 above)
Comment (gcor) [reply src/lib/components/ProjectsPanel/components/Modal.tsx]: added lib)

**Comment by gcor** [comment]
> It seems that there will be a lot of responsive rules for this component.
> 
> 💡 How about replacing paddings and margins in `px` with **em**?

I don't fully understand the benefits of em, rem, etc. This will add miscommunication with designs. Also not easy to work with small values, like padding\borders\ets. Maybe Im conservative))

**Comment by a-chabin** [comment]
> I don't fully understand the benefits of em, rem, etc. This will add miscommunication with designs. Also not easy to work with small values, like padding\borders\ets. Maybe Im conservative))

I think I'll say this:
soglasen, poka voobshe norm)))

**Comment by a-chabin** [comment]
Great! :)

I propose to improve this control in the next PRs :)

**Pull request merged**